hbase-server/src/main/java/org/apache/hadoop/hbase/mob/MobFileCleanerChore.java

```diff
@@ -52,6 +64,12 @@ public MobFileCleanerChore(HMaster master) {
     this.master = master;
     cleaner = new ExpiredMobFileCleaner();
     cleaner.setConf(master.getConfiguration());
+    int threadCount = master.getConfiguration().getInt(MobConstants.MOB_CLEANER_THREAD_COUNT,
+      MobConstants.DEFAULT_MOB_CLEANER_THREAD_COUNT);
+
+    ThreadFactory threadFactory =
+      new ThreadFactoryBuilder().setDaemon(true).setNameFormat("mobfile-cleaner-pool-%d").build();
+    threadPool = Executors.newFixedThreadPool(threadCount, threadFactory);
```

Comment:
If there is only one thread configured, we could make use of MoreExecutors.newDirectExecutorService, so we will not have an extra thread here?

Reply:
changed to MoreExecutors.newDirectExecutorService when single thread is configured